sycl/source/detail/platform_impl.cpp

```diff
@@ -303,9 +304,20 @@ static void filterDeviceFilter(vector_class<RT::PiDevice> &PiDevices,
   if (!FilterList)
     return;
 
+  // remember the last backend that has gone through this filter function
+  // to assign a unique device id number across platforms that belong to
+  // the same backend. For example, opencl:cpu:0, opencl:acc:1, opencl:gpu:2
+  static backend lastBackend = backend::all;
   backend Backend = Plugin.getBackend();
   int InsertIDx = 0;
-  int DeviceNum = 0;
+  // DeviceNums should be given consecutive numbers across platforms.
+  // So, we keep the device num for the successive calls to this function.
+  static int DeviceNum = 0;
```

Comment:
This is not thread-safe. Calls to .get_devices() from multiple threads would be broken.

Reply:
Do you have any suggestion on how to make it thread safe?